runtime-light/stdlib/confdata/confdata-functions.cpp

```diff
@@ -0,0 +1,92 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2024 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "runtime-light/stdlib/confdata/confdata-functions.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <string_view>
+#include <utility>
+
+#include "runtime-core/runtime-core.h"
+#include "runtime-core/utils/kphp-assert-core.h"
+#include "runtime-light/component/component.h"
+#include "runtime-light/coroutine/task.h"
+#include "runtime-light/stdlib/component/component-api.h"
+#include "runtime-light/tl/tl-core.h"
+#include "runtime-light/tl/tl-functions.h"
+#include "runtime-light/tl/tl-types.h"
+#include "runtime-light/utils/context.h"
+#include "runtime-light/utils/json-functions.h"
+
+namespace {
+
+constexpr const char *CONFDATA_COMPONENT_NAME = "confdata"; // TODO: it may actually have an alias specified in linking config
+
+mixed extract_confdata_value(tl::confdataValue &&confdata_value) noexcept {
+  if (confdata_value.is_php_serialized && confdata_value.is_json_serialized) { // check that we don't have both flags set
+    php_warning("confdata value has both php_serialized and json_serialized flags set");
+    return {};
+  }
+  if (confdata_value.is_php_serialized) {
+    php_critical_error("unimplemented"); // TODO
+  } else if (confdata_value.is_json_serialized) {
+    return f$json_decode(confdata_value.value);
+  } else {
+    return std::move(confdata_value.value);
+  }
+}
+
+} // namespace
+
+// TODO: the performance of this implementation can be enhanced. rework it when the platform has specific API for that
+bool f$is_confdata_loaded() noexcept {
+  auto &component_ctx{*get_component_context()};
+  if (const auto stream_d{component_ctx.open_stream(std::string_view{CONFDATA_COMPONENT_NAME})}; stream_d != INVALID_PLATFORM_DESCRIPTOR) {
+    return (component_ctx.release_stream(stream_d), true);
+  }
+  return false;
+}
+
+task_t<mixed> f$confdata_get_value(string key) noexcept {
+  tl::TLBuffer tlb{};
+  tl::ConfdataGet{.key = std::move(key)}.store(tlb);
+  const auto response{co_await f$component_client_fetch_response(
+    co_await f$component_client_send_request({CONFDATA_COMPONENT_NAME, static_cast<string::size_type>(std::string_view{CONFDATA_COMPONENT_NAME}.size())},
+                                             {tlb.data(), static_cast<string::size_type>(tlb.size())}))};
```

Comment:
May be split into some variables? It's really hard to read multiple `co_await` and `static_cast` in single statement.

Reply:
I prefer to not introduce additional variables if they are supposed to be used only once. At the same time, I understand that readability degrades a bit, so I can split this expression into variables if you insist.

Reply:
Yeah, I think it's overcomplicated

Reply:
Fixed